test_pull_requests

```diff
@@ -874,6 +890,83 @@ popd
       end
     end
 
+    # has_valid_flake_comment? determines if a valid flake explanation comment exists,
+    # if there exists no valid comment, a human-readable reason why to help users 
+    # understand the expected syntax
+    #
+    # A flake comment is valid iff:
+    #  - the comment contains the failed build URL
+    #  - all given issue links in the comment:
+    #    - reside in the correct repository
+    #    - point to issues with the correct label
+    def has_valid_flake_comment?(comments, build_url, repo, settings)
+      flake_issue_spec = /(https?:\/\/github.com\/#{Regexp.quote(Properties['github_user'])}\/#{Regexp.quote(settings['flake_repo'])}\/issues\/([1-9][0-9]*))/
+      
+      valid_comment_exists = false
+      reason = %{No comment matching the flake comment syntax was found. A correct comment should include a link to the failed Jenkins job as well as at least one link to a GitHub issue with the `kind/test-flake` label. An example of a comment with correct syntax would look like:
+```
+#{build_url}
+https://github.com/#{Properties['github_user']}/#{settings['flake_config']['flake_repo']}/issues/0
+```}
+
+      comments = sort_comments(comments)
+      comments.each do |comment|
+        comment_matches_syntax = comment['body'] =~ /#{Regexp.quote(build_url)}/ && comment['body'] =~ flake_issue_spec
+        # we can't allow the bot to be a valid author, or we would pick up
+        # comments where the bot explains the flake syntax to GitHub users
+        comment_author_valid = comment['user']['login'] != Properties['bot_github_user'] && user_trusted?(comment['user']['login'], repo, settings)
```

Comment:
Debating whether we care about the user being trusted.  The merge or test comment still has to be valid right.  So why not let an untrusted user reference the flake.